auditlog/management/commands/auditlogflush.py

```diff
@@ -8,27 +10,45 @@ class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument(
-            "-y, --yes",
+            "-y",
+            "--yes",
             action="store_true",
             default=None,
             help="Continue without asking confirmation.",
             dest="yes",
         )
+        parser.add_argument(
+            "-b",
+            "--before-date",
+            default=None,
+            help="Flush all entries with a timestamp before a given date (ISO 8601).",
+            dest="before_date",
+            type=lambda b: datetime.datetime.strptime(b, "%Y-%m-%d").date(),
+        )
 
     def handle(self, *args, **options):
         answer = options["yes"]
+        before = options["before_date"]
 
         if answer is None:
-            self.stdout.write(
+            warning_message = (
                 "This action will clear all log entries from the database."
             )
+            if before is not None:
+                warning_message = f"This actions will clear all log entries before {before} from the database."
+            self.stdout.write(warning_message)
             response = (
                 input("Are you sure you want to continue? [y/N]: ").lower().strip()
             )
             answer = response == "y"
 
         if answer:
-            count, _ = LogEntry.objects.all().delete()
+            entries = LogEntry.objects
+            if before is not None:
+                entries = entries.filter(timestamp__lt=before)
+            else:
+                entries = entries.all()
```

Comment:
I think you can remove the `else` condition and define `entries` as `LogEntry.objects.all()` directly.

Reply:
I'm not sure that I follow. Do you mean to suggest:
```python
entries = LogEntry.objects
if before is not None:
    entries = entries.filter(timestamp__lt=before)
count, _ = entries.delete()
```
or are you suggesting something else?

Reply:
You can write this:
```
entries = LogEntry.objects.all()
if before is not None:
    entries = entries.filter(timestamp__lt=before)
count, _ = entries.delete()
```

Reply:
TIL, Django query execution is even lazier than I thought.

Done.